libraries/botbuilder-azure/src/doOnce.ts

```diff
@@ -7,13 +7,15 @@
  */
 
 export class DoOnce<T> {
-    private task: Promise<T>;
+    private task: {
+        [dbAndContainerKey: string]: Promise<T>;
+    } = {};
 
-    public waitFor(fn: () => Promise<T>): Promise<T> {
-        if (!this.task) {
-            this.task = fn();
+    public waitFor(dbAndContainerKey: string, fn: () => Promise<T>): Promise<T> {
```

Comment:
There's nothing really specific in this class about databases and containers so how about just calling the variable `key`?